pkg/minishift/update/update_test.go

```diff
@@ -0,0 +1,72 @@
+/*
+Copyright (C) 2017 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package update
+
+import (
+	"fmt"
+	"math/rand"
+	"net/http"
+	"path/filepath"
+	"runtime"
+	"testing"
+	"time"
+)
+
+var (
+	archivePath string
+	err         error
+	port        int
+)
+
+func TestDownloadAndVerifyArchive(t *testing.T) {
+	port = randomNum(1000, 9999)
```

Comment:
Agree, as the current solution can still some up with a conflicting port. Randomization without check is not sufficient.